packages/cookie-consent/src/CookieConsentProvider/CookieConsentProvider.tsx

```diff
@@ -0,0 +1,227 @@
+
+  // We store unique categories names in an array
+  const categories = useMemo(
+    () =>
+      uniq([
+        ...(segmentIntegrations ?? []).map(({ category }) => category),
+      ]).sort(),
+    [segmentIntegrations],
+  )
+
+  // From the unique categories names we can now build our consent object
+  // and check if there is already a consent in a cookie
+  // Default consent if none is found is false
+  const cookieConsent = useMemo(
+    () =>
+      categories.reduce<Partial<Consent>>(
+        (acc, category) => ({
+          ...acc,
+          [category]: isConsentRequired
+            ? cookies?.[`${COOKIE_PREFIX}_${category}`] === 'true'
+            : true,
+        }),
+        {},
+      ),
+    [isConsentRequired, categories, cookies],
+  )
+
+  const saveConsent = useCallback(
+    (categoriesConsent: Partial<Consent>) => {
+      for (const [consentName, consentValue] of Object.entries(
+        categoriesConsent,
+      ) as [CategoryKind, boolean][]) {
+        const cookieName = `${COOKIE_PREFIX}_${consentName}`
+
+        if (!consentValue) {
+          // If consent is set to false we have to delete the cookie
+          document.cookie = cookie.serialize(cookieName, '', {
+            expires: new Date(0),
+          })
+        } else {
+          document.cookie = cookie.serialize(
+            `${COOKIE_PREFIX}_${consentName}`,
+            consentValue.toString(),
+            {
+              ...COOKIES_OPTIONS,
+              maxAge:
+                consentName === 'advertising'
+                  ? CONSENT_ADVERTISING_MAX_AGE
+                  : CONSENT_MAX_AGE,
+            },
+          )
+        }
+      }
+      // We set the hash cookie to the current consented integrations
+      document.cookie = cookie.serialize(
+        HASH_COOKIE,
+        integrationsHash.toString(),
+        {
+          ...COOKIES_OPTIONS,
+          // Here we use the shortest max age to force to ask again for expired consent
+          maxAge: CONSENT_ADVERTISING_MAX_AGE,
+        },
+      )
+      setNeedsConsent(false)
+    },
+    [integrationsHash],
+  )
+
+  const isSegmentAllowed = useMemo(
+    () =>
+      isConsentRequired
+        ? !needConsent &&
+          !!segmentIntegrations?.some(
+            integration => cookieConsent[integration.category],
+          )
+        : true,
+    [isConsentRequired, segmentIntegrations, cookieConsent, needConsent],
+  )
+
+  const segmentEnabledIntegrations = useMemo(
+    () => ({
+      All: false,
+      ...segmentIntegrations?.reduce(
+        (acc, integration) => ({
+          ...acc,
+          [integration.name]: cookieConsent[integration.category],
+        }),
+        {},
+      ),
+    }),
+    [cookieConsent, segmentIntegrations],
+  )
+
+  const value = useMemo(
+    () => ({
+      integrations,
+      needConsent,
+      isSegmentAllowed,
+      segmentIntegrations: segmentEnabledIntegrations,
```

Comment:
Don't think we need this in the context anymore

Reply:
It is still used in console.